### ISSUE TITLE
Protocol Analysis

### ISSUE THREAD

**Issue by samkristoff** [Digilent/openscope-mz#2]
Add protocol analysis for logic analyzer channels.

- UART
- SPI
- I2C
- Parallel Digital
- etc

**Comment by SimonFili** [comment]
Wondering where we would get the more bang for the buck (hours/$$)
1) Add protocol analysis to waveform live OR
2) Have a library for openscope for sigrok that already support a lot of protocol analysis ?

**Comment by avnrdf** [comment]
@SimonFili @samkristoff Please add sigrok support. Waveforms live is good enough for everything else, but the logic analyzer isn't very useful without a protocol decoder. 
While you could add this to Waveforms Live, Sigrok is regularly updated with tons of decoders and it would be a better idea to leverage that by adding Sigrok support.

**Comment by orzel** [comment]
I agree that Sigrok is (imho) the way to go for such kind of hardware. Please add Sigrok support.

**Comment by VictorTagayun** [comment]
Any progress on this?
Maybe the developer can point out on where in the code to start.

**Comment by rdragonrydr** [comment]
I have since bought one of these, but I still feel that the lack of protocol decoding (at least the very basics) is disappointing. This could be a really good all-in-one open hardware and software device, but it's missing what I feel is a core feature.

I'd love if Waveforms came with a simple set of UART, I2C, and SPI protocols, but Sigrok support would be nice too (and probably more useful and simpler to implement in the long run).
